File: runner/list_models.py
#!/usr/bin/env python3
"""
Output model names as JSON for use in CI workflows.
This keeps runner/models/__init__.py as the single source of truth.
"""
import json
import argparse
from runner.models import ALL_MODELS, ModelProvider, CIRunFrequency


def get_models(
    provider: ModelProvider | None = None,
    frequency: CIRunFrequency | None = None,
) -> list[str]:
    models = ALL_MODELS

    if provider is not None:
        models = [m for m in models if m.provider == provider]

    if frequency is not None:
        models = [m for m in models if m.ci_run_frequency == frequency]

    return [m.name for m in models]


def main():
    parser = argparse.ArgumentParser(description="List available models as JSON")
    parser.add_argument(
        "--provider",
        choices=["anthropic", "openai", "together", "google", "xai", "all"],
        default="all",
        help="Filter by provider (default: all)",
    )
    parser.add_argument(
        "--frequency",
        choices=["daily", "weekly", "monthly", "never", "all"],
        default="all",
        help="Filter by CI run frequency (default: all)",
    )
    parser.add_argument(
        "--format",
        choices=["json", "csv"],
        default="json",
        help="Output format (default: json)",
    )
    args = parser.parse_args()

    provider = None if args.provider == "all" else ModelProvider(args.provider)
    frequency = None if args.frequency == "all" else args.frequency
    models = get_models(provider, frequency)

    if args.format == "json":
        print(json.dumps(models))
    else:
        print(",".join(models))


if __name__ == "__main__":
    main()

File: runner/listModels.ts
#!/usr/bin/env bun
/**
 * Output model names as JSON for use in CI workflows.
 * Keeps runner/models/index.ts as the single source of truth.
 *
 * Usage:
 *   bun run runner/listModels.ts --frequency daily --format json
 */
import { ALL_MODELS, type CIRunFrequency } from "./models/index.js";

function getModels(frequency?: CIRunFrequency): string[] {
  let models = ALL_MODELS;
  if (frequency) models = models.filter((m) => m.ciRunFrequency === frequency);
  return models.map((m) => m.name);
}

function parseArgs(): { frequency?: string; format: string } {
  const args = process.argv.slice(2);
  let frequency: string | undefined;
  let format = "json";

  for (let i = 0; i < args.length; i++) {
    if (args[i] === "--frequency" && args[i + 1]) frequency = args[++i];
    if (args[i] === "--format" && args[i + 1]) format = args[++i];
  }

  return { frequency, format };
}

function main(): void {
  const { frequency, format } = parseArgs();

  const freq =
    frequency && frequency !== "all"
      ? (frequency as CIRunFrequency)
      : undefined;

  const models = getModels(freq);

  if (format === "json") {
    console.log(JSON.stringify(models));
  } else {
    console.log(models.join(","));
  }
}

main();
